src/main/java/com/cargohub/controllers/AuthenticationController.java

```diff
@@ -1,35 +1,80 @@
 package com.cargohub.controllers;
 
-import com.cargohub.dto.BillingDetailsDto;
 import com.cargohub.dto.UserDto;
+import com.cargohub.entities.RoleEntity;
+import com.cargohub.entities.UserEntity;
+import com.cargohub.models.AuthRequestModel;
+import com.cargohub.models.AuthResponseModel;
 import com.cargohub.models.RegistrationModel;
+import com.cargohub.security.jwt.JwtTokenProvider;
 import com.cargohub.services.UserService;
 import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.*;
-
-import java.util.ArrayList;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
 import java.util.List;
 
 @RestController
-@RequestMapping("/registration")
 public class AuthenticationController {
 
-    private final ModelMapper modelMapper;
+    private final AuthenticationManager authenticationManager;
+
+    private final JwtTokenProvider jwtTokenProvider;
+
     private final UserService userService;
 
-    public AuthenticationController(ModelMapper modelMapper, UserService userService) {
-        this.modelMapper = modelMapper;
+    private ModelMapper modelMapper;
+
+    @Autowired
+    public AuthenticationController(AuthenticationManager authenticationManager, JwtTokenProvider jwtTokenProvider, UserService userService, ModelMapper modelMapper) {
+        this.authenticationManager = authenticationManager;
+        this.jwtTokenProvider = jwtTokenProvider;
         this.userService = userService;
+        this.modelMapper = modelMapper;
     }
 
-    @PostMapping
+    @PostMapping("/registration")
     public ResponseEntity register(@RequestBody RegistrationModel registrationModel) {
         UserDto userDto = modelMapper.map(registrationModel, UserDto.class);
         userService.createUser(userDto);
 
         return new ResponseEntity(HttpStatus.CREATED);
     }
 
+    @PostMapping("/login")
+    public ResponseEntity login(@RequestBody AuthRequestModel requestDto) {
+        try {
+            String email = requestDto.getEmail();
+            authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(email, requestDto.getPassword()));
+            UserDto user = userService.getUser(email);
+
+            if (user == null) {
```

Comment:
Following throw is unnecessary, it duplicates the same in user service and is unreachable